fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/FlammableBlockRegistry.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1;
+
+import net.minecraft.block.Block;
+import net.minecraft.tag.Tag;
+
+import net.fabricmc.fabric.api.content.registry.v1.util.ContentRegistry;
+import net.fabricmc.fabric.impl.content.registry.FlammableBlockRegistryImpl;
+
+/**
+ * Registry of Blocks that can be set on fire.
+ */
+public interface FlammableBlockRegistry extends ContentRegistry<Block, FlammableBlockRegistry.Entry> {
+	/**
+	 * Create a new flammable block registry for a type of fire.
+	 * @param fireBlock the block that this registry belongs to
+	 * @param prefix the name of the block
+	 */
+	default FlammableBlockRegistry create(Block fireBlock, String prefix) {
```

Comment:
Javadoc should document thrown exception if the block is not a fire block.

Reply:
Wouldn't it be better to just only accept a `FireBlock`? It seems that Mojang is moving away from registering these sorts of things in code, so it's doubtful that the content registry for flammable blocks will ever need to expand.

Reply:
Many mods expose their block instances as `Block` such as vanilla. So you would need a cast in some cases, but it would also reduce the chance of error.

Reply:
I'm going to go for the documented throws approach